StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1006CodeFixProvider.cs

```diff
@@ -48,10 +49,15 @@ public override FixAllProvider GetFixAllProvider()
                 if (!hashToken.IsKind(SyntaxKind.HashToken))
                     continue;
 
-                SyntaxToken corrected = hashToken.WithoutTrailingWhitespace().WithoutFormatting();
-                Document updatedDocument = context.Document.WithSyntaxRoot(root.ReplaceToken(hashToken, corrected));
-                context.RegisterCodeFix(CodeAction.Create("Remove space", t => Task.FromResult(updatedDocument)), diagnostic);
+                context.RegisterCodeFix(CodeAction.Create("Remove space", t => GetTransformedDocument(context, root, hashToken)), diagnostic);
```

Comment:
:grey_exclamation: Move as much of the logic from this loop to the callback as possible.